### PR TITLE
Feature/mat 4260 measure bundle

### PR DESCRIPTION
**Pull request by nmorasb**
## MADiE PR

Jira Ticket: [MAT-4260](https://jira.cms.gov/browse/MAT-4260)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/cms/gov/madie/measure/resources/MeasureController.java [hunk 1]
```diff
@@ -124,6 +129,18 @@ public ResponseEntity<String> updateMeasure(
     return response;
   }
 
+  @GetMapping("/measures/{measureId}/groups")
+  public ResponseEntity<List<Group>> getGroups(@PathVariable String measureId) {
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/resources/MeasureController.java:133]: is this something unrelated to the story?

**Review by adongare — APPROVED** [review]

File: src/main/java/cms/gov/madie/measure/resources/MeasureController.java [hunk 2]
```diff
@@ -124,6 +129,18 @@ public ResponseEntity<String> updateMeasure(
...
+    return repository
+        .findById(measureId)
+        .map(
+            measure -> {
+              List<Group> groups = measure.getGroups() == null ? List.of() : measure.getGroups();
+              return ResponseEntity.ok(groups);
+            })
+        .orElseThrow(() -> new ResourceNotFoundException("Measure", measureId));
+  }
+
```
Comment (RohitKandimalla) [inline src/main/java/cms/gov/madie/measure/resources/MeasureController.java:143]: Why do we need this endpoint?

File: src/main/java/cms/gov/madie/measure/models/Measure.java [hunk 3]
```diff
@@ -62,6 +62,9 @@ public class Measure {
   private String measureName;
 
   private boolean active = true;
+  // TODO: determine if theres a way to set this from backend or if we should always trust user
+  // input for this field
+  private boolean cqlErrors;
```
Comment (RohitKandimalla) [inline src/main/java/cms/gov/madie/measure/models/Measure.java:67]: Either we need to validate CQL for both parsing and translation errors in the backEnd or trust.

I believe this will also be the case for test case execution results, we do the execution in UI and so once we get to a point where we have to save those results, then we have to trust UI or run the execution again.

**Review by RohitKandimalla — APPROVED** [review]

File: src/main/java/cms/gov/madie/measure/resources/MeasureController.java (hunk 2 above)
Comment (RohitKandimalla) [reply src/main/java/cms/gov/madie/measure/resources/MeasureController.java:143]: ok, Ashok already asked the same questions. nvm

File: src/main/java/cms/gov/madie/measure/resources/MeasureController.java (hunk 1 above)
Comment (nmorasb) [reply src/main/java/cms/gov/madie/measure/resources/MeasureController.java:133]: yeah this is unrelated to the story - I added it because I was in the controller, Brandon had been asking about an endpoint to fetch measure groups for testing purposes, and it was quick to add one. And as a subresource that we have endpoints exposed for creating and updating, having an endpoint to fetch the subresources also is not out of line with REST standards.

File: src/main/java/cms/gov/madie/measure/models/Measure.java (hunk 3 above)
Comment (nmorasb) [reply src/main/java/cms/gov/madie/measure/models/Measure.java:67]: Yeah, agreed. I just wanted to capture it for later discussion. I was also going to add a single tech debt item for this as well as to review other places were we implicitly trust all user input and see if there's anywhere we should stop doing that.

File: src/main/java/cms/gov/madie/measure/resources/MeasureController.java (hunk 1 above)
Comment (adongare) [reply src/main/java/cms/gov/madie/measure/resources/MeasureController.java:133]: got it. Thanks for the clarification!
